treeherder/services/max_runtime.py

```diff
@@ -0,0 +1,18 @@
+from datetime import datetime, timedelta
+from treeherder.perf.exceptions import MaxRuntimeExceeded
+
+
+class MaxRuntime:
+    DEFAULT_MAX_RUNTIME = timedelta(hours=23)
+
+    def __init__(
+        self,
+        max_runtime: timedelta = None,
+    ):
+        self.max_runtime = max_runtime or MaxRuntime.DEFAULT_MAX_RUNTIME
+
+    def quit_on_timeout(self, started_at):
```

Comment:
You could set `started_at` as an instance variable. And activate on demand; for example via a method such as `start_timer()`.